Verify Url causes the codegen process to fail

This fails due to:

Parsing: thelakes.page

Parsing: thelakes.tsuite

Parsing: thelakes.tcase

Cleaning folder: C:\Rockmelon\TheLakes\output-code

Creating file: 'PageDefinitions.cs'

Creating file: 'Pages\ThelakesPage.cs'

Creating file: 'TestCases\ThelakesTestCase.cs'

Creating file: 'TestSuites\ThelakesTestSuite.cs'

Creating file: 'Support/LocatorHelper.cs'

Creating file: 'Support/RmSingleCaseSuiteBase.cs'

Creating file: 'AutomationTests.csproj'

Creating file: 'Usings.cs'

Creating file: '.runsettings'

Creating file: '.code-metadata'

Executing: 'dotnet build'

Error: Command "dotnet build" failed:
MSBuild version 17.4.0+18d5aef85 for .NET
Determining projects to restore...
Restored C:\Rockmelon\TheLakes\output-code\AutomationTests.csproj (in 399 ms).
C:\Rockmelon\TheLakes\output-code\TestCases\ThelakesTestCase.cs(17,22): error CS1503: Argument 1: cannot convert from 'Microsoft.Playwright.IPage' to 'Microsoft.Playwright.ILocator' [C:\Rockmelon\TheLakes\output-code\AutomationTests.csproj]

Build FAILED.

C:\Rockmelon\TheLakes\output-code\TestCases\ThelakesTestCase.cs(17,22): error CS1503: Argument 1: cannot convert from 'Microsoft.Playwright.IPage' to 'Microsoft.Playwright.ILocator' [C:\Rockmelon\TheLakes\output-code\AutomationTests.csproj]
0 Warning(s)
1 Error(s)

Time Elapsed 00:00:01.42

![image](https://user-images.githubusercontent.com/114720205/233557868-ced983e1-4404-4929-8603-48c3a28cb0ac.png)

--- a/common/src/codegen/playwright-csharp-xunit/templates/BaseClasses.hbs
+++ b/common/src/codegen/playwright-csharp-xunit/templates/BaseClasses.hbs
@@ -38,6 +38,14 @@ public abstract class TestCaseBase : PageTest
     public TestCaseBase(PageTest testSuiteInstance) : base(testSuiteInstance)
     {
     }
+
+    /// <summary>
+    /// Custom Expect method that works from IFrameLocator
+    /// </summary>
+    public ILocatorAssertions Expect(IFrameLocator locator)
+    {
+        return Expect(locator.Locator(":root"));
+    }
 }
 
 /// <summary>

--- a/common/src/codegen/utils/locator-generator-dotnet/iframeId.ts
+++ b/common/src/codegen/utils/locator-generator-dotnet/iframeId.ts
@@ -3,6 +3,6 @@ import { escapeStr } from "../../utils/stringUtils";
 
 export default (params: ILocatorTemplateParam) => {
   return params.hasParams
-    ? `this._page.FrameLocator(string.Format(\"id=${escapeStr(params.locatorStr)}\", parameters))`
-    : `this._page.FrameLocator(\"id=${escapeStr(params.locatorStr)}\")`;
+    ? `this._page.FrameLocator(string.Format("[id=\\"${escapeStr(params.locatorStr)}\\"]", parameters))`
+    : `this._page.FrameLocator("[id=\\"${escapeStr(params.locatorStr)}\\"]")`;
 };

